crates/fuel-core/src/executor.rs

```diff
@@ -183,6 +190,32 @@ mod tests {
         }
     }
 
+    /// Bad transaction source: ignores the limit of `u16::MAX -1` transactions
+    /// that should be returned by [`TransactionsSource::next()`].
+    /// It is used only for testing purposes
+    pub struct BadTransactionsSource {
+        transactions: Mutex<RefCell<Vec<MaybeCheckedTransaction>>>,
```

Comment:
You either need to use `RefCell` or `Mutex`=D Both is too much=D

Reply:
Ah yes, I can access the data as mutable from the Lock, my bad. Fixed in 8d5e04c9de